consensus/hotstuff/eventloop/event_loop.go

```diff
@@ -33,17 +38,27 @@ var _ component.Component = (*EventLoop)(nil)
 
 // NewEventLoop creates an instance of EventLoop.
 func NewEventLoop(log zerolog.Logger, metrics module.HotstuffMetrics, eventHandler hotstuff.EventHandler, startTime time.Time) (*EventLoop, error) {
-	proposals := make(chan *model.Proposal)
-	quorumCertificates := make(chan *flow.QuorumCertificate, 1)
-	timeoutCertificates := make(chan *flow.TimeoutCertificate, 1)
+	// we will use a buffered channel to avoid blocking of caller
+	// we can't afford to drop messages since it undermines liveness, but we also want to avoid blocking of compliance
+	// engine. We assume that we should be able to process proposals faster than compliance engine feeds them, worst case
+	// we will fill the buffer and block compliance engine worker but that should happen only if compliance engine receives
+	// large number of blocks in short period of time(when catching up for instance).
+	proposals := make(chan *model.Proposal, 1000)
+	// we will use a buffered channel to avoid blocking of caller
+	// we will create at most one partial TC per view and only in recovery path
+	// since we don't expect a large number of messages we are using a buffered channel.
+	partialTcCreated := make(chan *hotstuff.PartialTcCreated, 10)
```

Comment:
I would like to add some visibility into these channels, now that they can store a significant backlog of messages. I'll add a note to the metrics issue.